### PR TITLE
Redux form

### PR DESCRIPTION
**Pull request by LauchaBo**
## Summary

- Added redux form

## Trello Card

https://trello.com/c/hcYtSytt/16-redux-iv-redux-form-todo-finger

### CODE REVIEW

File: tic-tac-toe/src/components/ContactForm/index.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { Field, reduxForm } from 'redux-form';
+
+const ContactForm = props => {
```
Comment (FrankIglesias) [inline tic-tac-toe/src/components/ContactForm/index.js]: use a function instead

File: tic-tac-toe/src/components/ContactForm/index.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      <div>
+        <label htmlFor="email">Email</label>
+        <Field name="email" component="input" type="email" />
+      </div>
+      <button type="submit">Submit</button>
+    </form>
+  );
+};
+export default reduxForm({
+  // a unique name for the form
```
Comment (FrankIglesias) [inline tic-tac-toe/src/components/ContactForm/index.js]: remove this comment

File: tic-tac-toe/src/components/ContactPage/index.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { connect } from 'react-redux';
+
+import ContactForm from '../ContactForm';
+
+class ContactPage extends React.Component {
+  submit = values => {
+    // print the form values to the console
```
Comment (FrankIglesias) [inline tic-tac-toe/src/components/ContactPage/index.js]: remove comment

File: tic-tac-toe/src/components/Game/index.js [hunk 4]
```diff
@@ -1,76 +1,59 @@
...
   };
 
-  handleClick(squareIndex) {
-    const history = this.state.history.slice(0, this.state.stepNumber + 1);
+  handleClick = squareIndex => {
+    const history = this.props.history.slice(0, this.props.stepNumber + 1);
     const current = history[history.length - 1];
-    const squares = current.squares.slice();
-    if (this.calculateWinner(squares) || squares[squareIndex]) {
+    const squares = [...current.squares.slice()];
```
Comment (FrankIglesias) [inline tic-tac-toe/src/components/Game/index.js]: remove the slices

File: tic-tac-toe/src/components/Game/index.js [hunk 5]
```diff
@@ -1,76 +1,59 @@
...
-      if (squares[a] && squares[a] === squares[b] && squares[a] === squares[c]) {
-        return squares[a];
-      }
-    }
-    return null;
-  }
+    squares[squareIndex] = this.props.xIsNext ? 'X' : 'O';
+    this.props.dispatch(
+      TicTacToeActions.editTile({
+        history: history.concat([
```
Comment (FrankIglesias) [inline tic-tac-toe/src/components/Game/index.js]: you can use a shorthand here like 
``` 
[...history, {squares}]
```

File: tic-tac-toe/src/components/Game/index.js [hunk 6]
```diff
@@ -80,4 +63,12 @@ class Game extends Component {
   }
 }
 
-export default Game;
+const mapStateToProps = store => {
+  debugger //eslint-disable-line
```
Comment (FrankIglesias) [inline tic-tac-toe/src/components/Game/index.js]: remove debugger

File: tic-tac-toe/src/index.js [hunk 7]
```diff
@@ -1,9 +1,23 @@
...
 
 import './scss/index.scss';
 import registerServiceWorker from './registerServiceWorker';
-import Game from './components/Game'; // eslint-disable-line import/first
+import Game from './components/Game';
+import ContactPage from './components/ContactPage';
+import store from './redux/store';
 
-ReactDOM.render(<Game />, document.getElementById('root'));
+// store.dispatch({ type: 'EDIT_TILE' });
```
Comment (FrankIglesias) [inline tic-tac-toe/src/index.js]: delete commented line

File: tic-tac-toe/src/components/ItemList/index.js [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    };
+  }
+
+  componentDidMount() {
+    this.fetchData('http://5826ed963900d612000138bd.mockapi.io/items');
+  }
+
+  fetchData(url) {
+    this.setState({ isLoading: true });
+    fetch(url)
```
Comment (FrankIglesias) [inline tic-tac-toe/src/components/ItemList/index.js]: move this to a file named
`src/services/itemService.js`

File: tic-tac-toe/src/index.js [hunk 9]
```diff
@@ -1,9 +1,23 @@
...
+
+const App = () => (
```
Comment (FrankIglesias) [inline tic-tac-toe/src/index.js]: use a function insatead of an arrow

**Review by FrankIglesias — CHANGES_REQUESTED** [review]

File: tic-tac-toe/src/components/ContactForm/index.js (hunk 1 above)
Comment (LauchaBo) [reply tic-tac-toe/src/components/ContactForm/index.js]: If I transform to a function it gets flooded with errors

**Review by FrankIglesias — APPROVED** [review]

**Comment by FrankIglesias** [comment]
Well done!

**Pull request merged**